### PR TITLE
Barnes Points

### PR DESCRIPTION
**Pull request by maddie-boisvert**
Implements a two versions of the Barnes Points system with support for Combined points, as well as the break down of Women's/Men's Sweep/Sculling points.

1. Traditional Barnes System: Events of the same boat class across all rower experience level are weighted the same. For example, a varsity 8+ and a novice 8+ will both be worth 30 points.

2. Weighted Barnes System: Varsity events are worth the full 100% of points for the event, Junior/2V events are 80%, Novice/Freshman/3V are worth 60%. This is the system used by Midwest Scholastic Rowing Association and Scholastic Rowing Association of Michigan.

### CODE REVIEW

File: src/calculators/BarnesPointsCalc.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+export type BarnesPointsResults = {
+  team: string;
+  points: number;
+  place: number;
+}[];
+
+export const maxPointsFromName = (eventName: string) => {
+  eventName = eventName.toUpperCase();
+  const boatClassCaptureExpression = /.* ([1248]+)[x\-\+]+/;
```
Comment (glenne) [inline src/calculators/BarnesPointsCalc.ts:15]: I think you don't need the + after each[] pattern as you don't want to allow more than digit or suffix.

File: src/components/SimpleTeamPoints.tsx
Comment (glenne) [inline src/components/SimpleTeamPoints.tsx:1]: I'm assuming the changes to this file are to get something quick up and running?  Given the name BarnesSimple I guess that's it.  I think you'll want to create a BarnesPoints.tsx file for your react view to make it 'pretty'.  See ACATeamPoints.tsx for an example.

**Review by glenne — COMMENTED** [review]
Great start Maddie!

If there is an existing regatta you would like to use as a test data source let me know and I'll export the result data for you. You can also create a test regatta with just a few entries if that helps and I can export that.

File: src/calculators/BarnesPointsCalc.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  if (!scalars || place > scalars.length) {
+    return 0;
+  }
+
+  return scalars[place - 1];
+}
+
+const trimCrewName = (crewName: string) => {
+  crewName = crewName.toUpperCase().trim();
+  const suffixExpression = /(.*) .$/;
```
[the remark this answers is not in the document]
Comment (maddie-boisvert) [reply src/calculators/BarnesPointsCalc.ts:65]: Correct, thats the intent. I often see team entered with A or B designations which should have points attributed to the same larger team. And yes, `replace` sounds like a much nicer option!

File: src/components/SimpleTeamPoints.tsx
Comment (maddie-boisvert) [reply src/components/SimpleTeamPoints.tsx:1]: Correct, just wanted to get something up to be sanity checked before splitting into a more customized Barnes component.

**Comment by maddie-boisvert** [comment]
Could we export the results for regattas r12645 and r12646? I already have full points calculated for them and can use them as regression tests.

File: src/calculators/BarnesPointsCalc.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    }
+    if (boatClass == 4) {
+      return 20;
+    }
+    if (boatClass == 8) {
+      return 30;
+    }
+  }
+
+  return 0;
```
Comment (glenne) [inline src/calculators/BarnesPointsCalc.ts:34]: I have a similar function in CrewTimerUtils that returns the number of seats - numSeatsFromName().  In mine I return 1 if the boat class is not recognized on the assumption there is at least one rower on board.

**Pull request merged**